server/app/repository/ApplicationEventRepository.java

```diff
@@ -111,10 +113,57 @@ public CompletionStage<ApplicationEventModel> insertStatusEvent(
         executionContext.current());
   }
 
+  /**
+   * Updates the ApplicationEvents and the Applications table to the latest status in a single
+   * transaction for a given list of applications.
+   *
+   * <p>Note - Application code must change both tables at once as we want to avoid inconsistency
+   * between the tables.
+   */
+  public void insertStatusEvents(
+      ImmutableList<ApplicationModel> applications,
+      Optional<AccountModel> optionalAdmin,
+      ApplicationEventDetails.StatusEvent newStatusEvent) {
+    ApplicationEventDetails details =
+        ApplicationEventDetails.builder()
+            .setEventType(ApplicationEventDetails.Type.STATUS_CHANGE)
+            .setStatusEvent(newStatusEvent)
+            .build();
+
+    List<Long> applicationIds =
+        applications.stream().map(app -> app.id).collect(Collectors.toList());
+
+    var applicationsStatusEvent =
+        applications.stream()
+            .map(app -> new ApplicationEventModel(app, optionalAdmin, details))
+            .collect(ImmutableList.toImmutableList());
+    try (Transaction transaction = database.beginTransaction(TxIsolation.SERIALIZABLE)) {
```

Comment:
Question: Does there need to be a corresponding "catch" block?

Reply:
Great question. We are starting a transaction not necessarily throwing an exception. We are signaling play that every sql query in this block should happen in one go and in a serial manner. The control should not be dropped halfway through the block. I liked this [blog](https://www.thenile.dev/blog/transaction-isolation-postgres) explaining this.